soda_machine.rb

```diff
@@ -7,12 +7,20 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.size
   end
 
   def find_soda(soda_brand)
+    @sodas.find{|soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
+    this_soda = find_soda(soda_brand)
+    if this_soda != nil
+      @sodas.delete_at(@sodas.index(this_soda))
```

Comment:
(@amgando) @salogel42 Oh goodness me. You're right. I'm almost a bit embarrassed now. :flushed: 

I tripped up when considering how different strings have different object ids even if they are the same string - yet they all `a` strings get deleted from `arr = ['a','a','b','a'] by `arr.delete('a')` - of course on reflection this a false equivalency because I'm comparing three different STRING objects with a value of 'a' to distinct Soda objects with their own distinct values. 

In short, I'd gotten in the habit of deleting by index likely because there was a point _at some time_ where it was somehow necessary and incorrectly learned a blanket rule.